### PR TITLE
Optional override

### LINKED ISSUE #15: Optional GasPrice in swap/mint/etc
Goerli was slow to confirm txs without bumping the gas price for the swap example (PR #14).  as a result, i added a quick hack to just pass a configurable gasPrice to the overrides.  This should be optional and needs to be fixed properly.

### PR DESCRIPTION
**Pull request by jrhea**
Replacing gasPrice param with an optional overrides param for flexibility

resolves #15

### CODE REVIEW

**Review by pvienhage — APPROVED** [review]
Looks good, def prefer direct passing of overrides

File: src/swap.ts [hunk 1]
```diff
@@ -70,7 +71,7 @@ export async function swap(
   amount: BigNumber,
   kind: SwapKind,
   limit: BigNumber,
-  gasPrice: BigNumber,
+  overrides: PayableOverrides | undefined,
```
Comment (sentilesdal) [inline src/swap.ts:74]: for optional it's more semantic to do `overrides?: PayableOverrides`

File: src/swap.ts [hunk 2]
```diff
@@ -105,20 +106,12 @@ export async function swap(
...
-  const swapReceipt = await vaultContract.swap(
-    swap,
-    funds,
-    limit,
-    deadline,
-    overrides
-  );
+
+  const swapReceipt =
+    overrides == undefined
```
Comment (sentilesdal) [inline src/swap.ts:112]: `===` is safer than `==` in JS

File: src/swap.ts [hunk 3]
```diff
@@ -105,20 +106,12 @@ export async function swap(
...
+      ? await vaultContract.swap(swap, funds, limit, deadline)
+      : await vaultContract.swap(swap, funds, limit, deadline, overrides);
```
Comment (sentilesdal) [inline src/swap.ts:114]: can you not just pass undefined for overrides?

**Review by sentilesdal — APPROVED** [review]

File: src/swap.ts (hunk 3 above)
Comment (jrhea) [reply src/swap.ts:114]: ya, for swap i can, but not for mint

Comment (jrhea) [reply src/swap.ts:114]: swap gives an error when passing undefined

**Pull request merged**